apps/passport-server/test/user/testUserSync.ts

```diff
@@ -1,53 +1,497 @@
+    application.expressContext.localEndpoint,
+    encryptionKey
+  );
+  expect(initialLoadResult.success).to.eq(false);
+  expect(initialLoadResult.error?.name).to.eq("NotFound");
+
+  // Create storage
+  const uploadResult1 = await requestUploadEncryptedStorage(
     application.expressContext.localEndpoint,
     encryptionKey,
-    encryptedData
+    encryptedData1
   );
+  expect(uploadResult1.error).to.eq(undefined);
+  expect(uploadResult1.success).to.eq(true);
+  expect(uploadResult1.value).to.not.be.undefined;
+  expect(uploadResult1.value?.revision).to.not.be.undefined;
+  const rev1 = uploadResult1.value?.revision;
+
+  await fetchAndCheckStorage(application, encryptionKey, rev1, plaintextData1);
 
-  expect(uploadResult.error).to.eq(undefined);
-  expect(uploadResult.success).to.eq(true);
+  const plaintextData2 = {
+    test: "test",
+    two: 2
+  };
+
+  const encryptedData2 = await passportEncrypt(
+    JSON.stringify(plaintextData2),
+    encryptionKey
+  );
 
-  const secondLoadResult = await requestEncryptedStorage(
+  // Update storage
+  const uploadResult2 = await requestUploadEncryptedStorage(
     application.expressContext.localEndpoint,
+    encryptionKey,
+    encryptedData2
+  );
+  expect(uploadResult2.error).to.eq(undefined);
+  expect(uploadResult2.success).to.eq(true);
+  expect(uploadResult2.value).to.not.be.undefined;
+  expect(uploadResult2.value?.revision).to.not.be.undefined;
+  const rev2 = uploadResult2.value?.revision;
+  expect(rev2).to.not.eq(rev1);
+
+  await fetchAndCheckStorage(application, encryptionKey, rev2, plaintextData2);
+}
+
+export async function testUserSyncWithRev(application: Zupass): Promise<void> {
+  const crypto = await PCDCrypto.newInstance();
+  const encryptionKey = await crypto.generateRandomKey();
+
+  const plaintextData1 = {
+    test: "test",
+    one: 1
+  };
+
+  const encryptedData1 = await passportEncrypt(
+    JSON.stringify(plaintextData1),
     encryptionKey
   );
 
-  if (secondLoadResult.value == null) {
-    throw new Error("expected to be able to load e2ee");
-  }
+  // Storage shouldn't exist yet.
+  const initialLoadResult = await requestEncryptedStorage(
+    application.expressContext.localEndpoint,
+    encryptionKey,
+    "0"
+  );
+  expect(initialLoadResult.success).to.eq(false);
+  expect(initialLoadResult.error?.name).to.eq("NotFound");
 
-  const decrypted: string = await passportDecrypt(
-    secondLoadResult.value,
+  // Attempting update with base rev will result in NotFound, not a conflict.
+  const notfoundUploadResult = await requestUploadEncryptedStorage(
+    application.expressContext.localEndpoint,
+    encryptionKey,
+    encryptedData1,
+    "0"
+  );
+  expect(notfoundUploadResult.value).to.eq(undefined);
+  expect(notfoundUploadResult.success).to.eq(false);
+  expect(notfoundUploadResult.error?.name).to.eq("NotFound");
+
+  // Create storage with no base rev (required for first write)
```

Comment:
probably worth changing all shorthand `baseRevision` (i.e. forms like `base rev`) to `known rev` or equivalent